2015-Fall/oPolo/Calculator/Calculator/ViewController.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by Jonas Bruun Jacobsen on 11/07/15.
+//  Copyright (c) 2015 Jonas Bruun Jacobsen. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    
+    @IBOutlet weak var display: UILabel!
+    @IBOutlet weak var typingHistory: UILabel!
+    
+    var userIsInTheMiddleOfTypingANumber: Bool = false
+    
+    var brain = CalculatorBrain()
+
+    override func viewDidLoad() {
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "updateTypingHistory", name: "QueuedCalculationsChanged", object: nil)
```

Comment:
Now I see where you were going with this. Although it is a valid solution it is also a bit unnecessary in this specific case. Why not call ``updateTypingHistory()`` yourself when appending/operating?.

Additionally, static strings like ``"QueuedCalculationsChanged"`` should be a constant that you use both in ``postNotificationName`` and in ``addObserver``  instead. This is a bit more safe - especially when refactoring.